From e103e1e1430ad0d8b99108fe0e943c967104e9a7 Mon Sep 17 00:00:00 2001
From: Marian-Cristian Rotariu <marian.rotariu@freescale.com>
Date: Thu, 22 Oct 2015 12:47:11 +0300
Subject: [PATCH 105/245] dpaa_eth: add primitives for buffer alignment

These primitives are used on ARM architectures that have the 4K boundary
limitation.

Signed-off-by: Marian-Cristian Rotariu <marian.rotariu@freescale.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h
index 40d956b..056ac82 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h
@@ -661,4 +661,11 @@ static inline void _dpa_bp_free_pf(void *addr)
 	put_page(virt_to_head_page(addr));
 }
 
+#ifdef CONFIG_ARM64
+#define HAS_DMA_ISSUE(start, size) \
+	(((u64)(start) ^ ((u64)(start) + (u64)(size))) & ~0xFFF)
+
+#define BOUNDARY_4K(start, size) (((u64)(start) + (u64)(size)) & ~0xFFF)
+#endif
+
 #endif	/* __DPA_H */
-- 
2.1.0.27.g96db324

